pmda/parallel.py

```diff
@@ -307,144 +367,123 @@ def run(self,
-                wait_start = time.time()
-                res = blocks.compute(**scheduler_kwargs)
-            # hack to handle n_frames == 0 in this framework
-            if len(res) == 0:
-                # everything else wants list of block tuples
-                res = [([], [], [], 0, wait_start, 0, 0)]
-            # record conclude time
-            with timeit() as conclude:
-                self._results = np.asarray([el[0] for el in res])
-                # save the frame numbers for all blocks
-                self._blocks = _blocks
-                self._conclude()
-        # put all time information into the timing object
-        self.timing = Timing(
-            np.hstack([el[1] for el in res]),
-            np.hstack([el[2] for el in res]), total.elapsed,
-            np.array([el[3] for el in res]), time_prepare,
-            conclude.elapsed,
-            # waiting time = wait_end - wait_start
-            np.array([el[4]-wait_start for el in res]),
-            np.array([el[5] for el in res]),
-            np.array([el[6] for el in res]))
         return self
 
-    def _dask_helper(self, bslice, indices, top, traj):
-        """helper function to actually setup dask graph"""
+    def _map_chunk(self, bslice):
+        """function to setup chunk dask graph"""
         # wait_end needs to be first line for accurate timing
         wait_end = time.time()
-        # record time to generate universe and atom groups
-        with timeit() as b_universe:
-            u = mda.Universe(top, traj)
-            agroups = [u.atoms[idx] for idx in indices]
-
         res = []
         times_io = []
         times_compute = []
         # NOTE: bslice.stop cannot be None! Always make sure
         #       that it comes from  _trajectory.check_slice_indices()!
         for i in range(bslice.start, bslice.stop, bslice.step):
+            self._frame_index = i
             # record io time per frame
             with timeit() as b_io:
                 # explicit instead of 'for ts in u.trajectory[bslice]'
                 # so that we can get accurate timing.
-                ts = u.trajectory[i]
+                self._ts = self._trajectory[i]
             # record compute time per frame
             with timeit() as b_compute:
-                res = self._reduce(res, self._single_frame(ts, agroups))
+                res = self._reduce(res, self._single_frame())
             times_io.append(b_io.elapsed)
             times_compute.append(b_compute.elapsed)
 
         # calculate io and compute time per block
         return np.asarray(res), np.asarray(times_io), np.asarray(
-            times_compute), b_universe.elapsed, wait_end, np.sum(
+            times_compute), wait_end, np.sum(
             times_io), np.sum(times_compute)
 
     @staticmethod
     def _reduce(res, result_single_frame):
         """ 'append' action for a time series"""
         res.append(result_single_frame)
         return res
+
+    def __getstate__(self):
```

Comment:
Does DaskMixin require the whole class to be picklable?

Reply:
Yes...I mean in the old implementation, the whole class has to be picklable as well.

FYI, the code here is not needed anymore after https://github.com/MDAnalysis/mdanalysis/pull/2893 is merged